src/NimBLEClient.cpp

```diff
@@ -552,6 +552,21 @@ uint16_t NimBLEClient::getConnId() {
 } // getConnId
 
 
+/**
+ * @brief Set the connection id for this client.
+ * @param [in] conn_id The connection id.
+ * @note Sets the connection established flag to true.
+ * @note This is designed to be used when a connection is made outside of the
+ *       NimBLEClient class, such as when a connection is made by the
+ *       NimBLEServer class and the client is passed the connection id. This use
+ *       enables the GATT Server to read the name of the device that has
+ *       connected to it.
+ */
+void NimBLEClient::setConnId(uint16_t conn_id) {
+    m_conn_id = conn_id;
```

Comment:
I think we need to add a check here to make sure the client instance is not already connected to a peer and return false if it is. We should also set the peer address here as well which can be retrieved with `ble_gap_conn_find` using the conn_id, or we can add it as another parameter to the function.

Reply:
The check for `isConnected()` would prevent the `Client` object from being reused with a server, since once `setConnId(...)` is called, `isConnected()` would return true. There would be no way to then reset that state. Would you be open to adding a `reset()` function or something similar which clears the `m_conn_id`, `m_connEstablished`, and `m_peerAddress` members for this use case?

Reply:
Had the same thought and decided to make that update :)

I did end up renaming it to `setConnection(...)` overloads. Not sure _why_ you'd pass anything other than the `NimBLEConnInfo`, but I think there may be use-cases.

Regarding your question, it was merely to remove the need for multiple allocations / deallocations for what I figured was a pretty simple use-case. That being said, I'm not opposed to going that route.

Reply:
> Regarding your question, it was merely to remove the need for multiple allocations / deallocations for what I figured was a pretty simple use-case. That being said, I'm not opposed to going that route.

I would say we should create a clearConnection function in this case, couldn't hurt.